glyphs2fontir/src/erase_open_corners.rs

```diff
@@ -0,0 +1,524 @@
+    for candidate in iter_erasure_candidates(path.elements(), is_closed) {
+        if !candidate.points_are_right_of_line() {
+            continue;
+        }
+        let Some(Intersection { t0, t1, .. }) = candidate.intersection() else {
+            continue;
+        };
+        // invert value of t0 so for both values '0' means at the open corner
+        // <https://github.com/googlefonts/glyphsLib/blob/74c63244fdbef1da5/Lib/glyphsLib/filters/eraseOpenCorners.py#L105>
+        let t0_inv = 1.0 - t0;
+        if ((t0_inv < 0.5 && t1 < 0.5) || t0_inv < 0.3 || t1 < 0.3)
+            && t1 > 0.0001
+            && t0_inv > 0.0001
+        {
+            let ix = candidate.ix;
+            todo.extend([
+                (ix, SegmentOp::Replace(candidate.prev.subsegment(0.0..t0))),
+                (ix + 1, SegmentOp::Delete),
+                (
+                    ix + 2,
+                    SegmentOp::Replace(candidate.next.subsegment(t1..1.0)),
+                ),
+            ]);
+        }
+    }
+    if todo.is_empty() {
+        return None;
+    }
+
+    // it will be simpler to operate on segments:
+    let segments = kurbo::segments(path.elements().iter().copied()).collect::<Vec<_>>();
+    // and then we will make an equal number of ops as segments, also to simplify:
+    let mut ops = vec![SegmentOp::Retain; segments.len()];
+    for (ix, op) in todo {
+        // ix isn't always right because we didn't know the number of segments
+        // before, and we handle the first two points at the end
+        ops[ix % segments.len()] = op;
+    }
+
+    let mut els = Vec::with_capacity(path.elements().len());
+    let mut iter = segments
+        .iter()
+        .zip(ops)
+        .filter_map(|(seg, op)| op.apply(*seg))
+        .peekable();
+
+    // we will use `PathSeg::as_path_el` below, but that doesn't handle moveto
+    els.push(PathEl::MoveTo(iter.peek().unwrap().start()));
+    els.extend(iter.map(|seg| seg.as_path_el()));
+    if is_closed {
+        els.push(PathEl::ClosePath);
+    }
+    Some(BezPath::from_vec(els))
+}
+
+#[derive(Clone, Debug)]
+enum SegmentOp {
+    Retain,
+    Delete,
+    Replace(PathSeg),
+}
+
+impl SegmentOp {
+    fn apply(self, seg: PathSeg) -> Option<PathSeg> {
+        match self {
+            SegmentOp::Retain => Some(seg),
+            SegmentOp::Delete => None,
+            SegmentOp::Replace(seg) => Some(seg),
+        }
+    }
+}
+
+/// A potentially open corner.
+///
+/// This is any line segment, along with the preceding and following path segments.
+struct ErasureCandidate {
+    // index of the *first* participating segment, which contains prev_pt
+    ix: usize,
+    seg: Line,
+    prev: PathSeg,
+    next: PathSeg,
+}
+
+impl ErasureCandidate {
+    //https://github.com/googlefonts/glyphsLib/blob/74c63244fdb/Lib/glyphsLib/filters/eraseOpenCorners.py#L66-L71
+    fn points_are_right_of_line(&self) -> bool {
```

Comment:
I don't find this methods behavior very intuitive. I suggest copying over this comment:

```python
            # Are the incoming point from the previous segment and the outgoing point
            # from the next segment both on the right side of the line?
            # (see discussion at https://github.com/googlefonts/glyphsLib/pull/663)
```